pkg/pillar/kubeapi/vitoapiserver.go

```diff
@@ -313,6 +313,89 @@ func RolloutDiskToPVC(ctx context.Context, log *base.LogObject, exists bool,
 	return nil
 }
 
+// GetPVFromPVC : Returns volume name (PV) from the PVC name
+func GetPVFromPVC(pvcName string, log *base.LogObject) (string, error) {
+	// Get the Kubernetes clientset
+	clientset, err := GetClientSet()
+	if err != nil {
+		err = fmt.Errorf("failed to get clientset: %v", err)
+		log.Error(err)
+		return "", err
+	}
+	// Get the PersistentVolumeClaim (PVC)
+	pvc, err := clientset.CoreV1().PersistentVolumeClaims(EVEKubeNameSpace).Get(context.TODO(), pvcName, metav1.GetOptions{})
+	if err != nil {
+		log.Errorf("Error fetching PersistentVolumeClaim %s: %v", pvcName, err)
```

Comment:
You could use the same approach from line [321](https://github.com/lf-edge/eve/pull/4359/files#diff-01e7f32d4e5ab78e4a798e525ecb0ba7529ba505aebd89c3cdbea6fb32ebf333R321) and return the same logged error.

Reply:
Ok I will make those changes across all 3 API.

Reply:
Amended the commit